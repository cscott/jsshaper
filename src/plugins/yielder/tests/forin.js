var log = (typeof console !== "undefined") && console.log || print;
(typeof require !== 'undefined') && require('setup.js');

function foo(b) {
    try {
        yield 1;
        if (b) throw b;
        yield 2;
    } finally {
        log('finally!');
    }
}

for (var x in foo()) {
    log('a');
    log(x);
    log('b');
    break;
}
log('c');

for (var x in foo()) {
    log(x);
}

try {
    for (var x in foo(true)) {
        log(x);
    }
} catch (e) {
    log(e);
}
