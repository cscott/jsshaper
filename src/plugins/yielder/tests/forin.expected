a
1
b
finally!
c
1
2
finally!
1
finally!
true
